apps/web/src/app/auth/settings/security/change-kdf/change-kdf.component.ts

```diff
@@ -24,8 +26,29 @@ import { ChangeKdfConfirmationComponent } from "./change-kdf-confirmation.compon
 })
 export class ChangeKdfComponent implements OnInit {
   kdfConfig: KdfConfig = DEFAULT_KDF_CONFIG;
-  kdfType = KdfType;
   kdfOptions: any[] = [];
+  private destroy$ = new Subject<void>();
+
+  protected formGroup = new FormGroup({
+    kdf: new FormControl(KdfType.PBKDF2_SHA256, [Validators.required]),
+    kdfConfig: new FormGroup({
+      iterations: new FormControl(this.kdfConfig.iterations, [
+        Validators.required,
+        Validators.min(PBKDF2_ITERATIONS.min),
+        Validators.max(PBKDF2_ITERATIONS.max),
+      ]),
+      memory: new FormControl(null, [
+        Validators.required,
+        Validators.min(ARGON2_MEMORY.min),
+        Validators.max(ARGON2_MEMORY.max),
+      ]),
+      parallelism: new FormControl(null, [
+        Validators.required,
+        Validators.min(ARGON2_PARALLELISM.min),
+        Validators.max(ARGON2_PARALLELISM.max),
+      ]),
+    }),
+  });
```

Comment:
You need to import the formBuilder in the constructor and use that. It should look something like:

```suggestion
    protected formGroup = this.formBuilder.group({
    kdf: [KdfType.PBKDF2_SHA256, [Validators.required]],
    kdfConfig: this.formBuilder.group({
      iterations: [this.kdfConfig.iterations, [
        Validators.required,
        Validators.min(PBKDF2_ITERATIONS.min),
        Validators.max(PBKDF2_ITERATIONS.max),
      ]],
      memory: [null as number | null, [
        Validators.required,
        Validators.min(ARGON2_MEMORY.min),
        Validators.max(ARGON2_MEMORY.max),
      ]],
      parallelism: [null as number | null, [
        Validators.required,
        Validators.min(ARGON2_PARALLELISM.min),
        Validators.max(ARGON2_PARALLELISM.max),
      ]],
    }),
  });
```